presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorAccessControl.java

```diff
@@ -195,6 +195,7 @@ default void checkCanSelectFromColumns(ConnectorTransactionHandle transactionHan
      *
      * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
      */
+    @Deprecated
     default void checkCanSelectFromTable(ConnectorTransactionHandle transactionHandle, Identity identity, SchemaTableName tableName)
     {
         denySelectTable(tableName.toString());
```

Comment:
This should delegate to `checkCanSelectFromColumns`. Then subclasses can delete their implementations of the deprecated methods after implementing the column level checks.